contracts/ENSGovernorV2.sol

```diff
@@ -0,0 +1,159 @@
+        proposalId = propose(targets, values, calldatas, description);
+
+        token.safeTransferFrom(msg.sender, timelock(), bondAmount);
+        createBond(proposalId, bondAmount);
+    }
+
+    /**
+     * @notice Calculate the bond amount based on the count of targets.
+     * @param countOfTargets The number of targets in the proposal.
+     * @return The calculated bond amount.
+     */
+    function calculateBond(
+        uint256 countOfTargets
+    ) public view returns (uint256) {
+        return countOfTargets * bondPricePerTarget;
+    }
+
+    /**
+     * @notice Create a bond for a proposal.
+     * @param proposalId The ID of the proposal.
+     * @param bondAmount The amount of bond.
+     */
+    function createBond(uint256 proposalId, uint256 bondAmount) internal {
+        lockedBondsBalance += bondAmount;
+
+        _proposalBonds[proposalId] = ProposalBond({
+            amount: bondAmount,
+            proposer: msg.sender,
+            refunded: false,
+            forfeited: false
+        });
+
+        emit BondCreated(msg.sender, bondAmount);
+    }
+
+    /**
+     * @notice Refund the bond for a proposal if conditions are met.
+     * @param proposalId The ID of the proposal.
+     */
+    function refundBond(uint256 proposalId) public {
+        ProposalState status = state(proposalId);
+        require(
+            status != ProposalState.Pending && status != ProposalState.Active,
+            "Wrong proposal status"
+        );
+
+        (uint256 againstVotesWithoutBond,uint256 againstVotes,,) = proposalVotes(proposalId);
+
+        require(
+            againstVotes >= againstVotesWithoutBond,
+            "Bond cannot be refunded"
+        );
+
+        ProposalBond storage bond = _proposalBonds[proposalId];
+
+        require(bond.proposer != address(0) , "Bond is not active");
+        require(!bond.refunded, "Bond is refunded");
+
+        token.safeTransferFrom(timelock(), bond.proposer, bond.amount);
+        lockedBondsBalance -= bond.amount;
+        bond.refunded = true;
+    }
+
+    /**
+     * @notice Check if we can forfeit bond of the proposal
+     * @param proposalId The ID of the proposal.
+     */
+    function checkAndForfeitBond(uint256 proposalId) public {
+        ProposalState status = state(proposalId);
+        require(
+            status != ProposalState.Pending && status != ProposalState.Active,
+            "Wrong proposal status"
+        );
+
+        (uint256 againstVotesWithoutBond,uint256 againstVotes,,) = proposalVotes(proposalId);
+
+        require(
+            againstVotes < againstVotesWithoutBond,
+            "Bond cannot be forfeited"
+        );
+
+        ProposalBond storage bond = _proposalBonds[proposalId];
+
+        require(bond.proposer != address(0) , "Bond is not active");
+        require(!bond.forfeited, "Bond already forfeited");
+
+        lockedBondsBalance -= bond.amount;
+        forfeitedBondsBalance += bond.amount;
+        bond.forfeited = true;
```

Comment:
If funds are held by the Governor instead of the Timelock, then they can be transferred *to* the timelock at the time of forfeiture to put them under the purview of the DAO.